File: math/primenumbers/151-reverse-words-in-a-string/reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let split=s.split(' ')

    let reversed=split.reverse()
    let res=""
    for(let i=0;i<reversed.length;i++){
        if(reversed[i].length>0){
            res+=reversed[i]+" "
        }
    }
    console.log(res)
    return res.trim()
};

File: math/primenumbers/151-reverse-words-in-a-string/reverse-words-in-a-string.java
class Solution {


    public String reverseWords(String s) {

        String[] s1=s.split(" ");

        StringBuilder sb=new StringBuilder("");

        int n=s1.length;
        System.out.println(Arrays.toString(s1));

        for(int i=n-1;i>=0;i--){
            if(!s1[i].equals("")){
                sb.append(s1[i]);
                sb.append(" ");
            }
        }

        
        System.out.println(sb);
        return sb.toString().trim();
    }
}